spec/take_away_ui_spec.rb

```diff
@@ -0,0 +1,232 @@
+    end
+
+    describe '#take_order' do
+
+      before do
+        allow(dummy_order).to receive(:add_to_basket)
+      end
+
+
+      it 'instantiates a new order' do
+        expect(dummy_order_klass).to receive(:new).and_return(dummy_order)
+        menu.take_order
+      end
+
+      it 'adds the order to basket' do
+        expect(dummy_order).to receive(:add_to_basket)
+        menu.take_order
+      end
+
+    end
+
+  describe '#select_dish' do
+      it {is_expected.to respond_to(:select_dish).with(2).arguments}
+
+
+      before do
+        allow(dummy_basket).to receive(:selection).and_return([[selection1]])
+        allow(dummy_basket_klass).to receive(:new).with(menu.selection, default_menu_dinner_mains).and_return(dummy_basket)
+        menu.select_dish("Cake", 3)
+      end
+
+      it 'puts dish selections into selection1 array' do
+        expect(menu.selection).to include selection1
+      end
+
+      it 'instantiates a new basket on first selection' do
+        expect(menu.basket).to eq dummy_basket
+      end
+
+      it 'puts dish selections into selection array' do
+        menu.select_dish("Tiramisu", 1)
+        expect(menu.basket.selection).to include selection2
+      end
+
+      it 'does not instantiate a new basket on second selections' do
+        menu.select_dish("Tiramisu", 1)
+        expect(menu.basket).to eq dummy_basket
+      end
+
+      it 'selecting dishes increases the size of selection array' do
+        expect {menu.select_dish("Tiramisu", 1)}.to change{menu.basket.selection.size}.by(1)
+      end
+
+
+
+      describe "#review_order" do
+        let(:sample_bill) {double :sample_bill}
+        let(:sample_total) {double :sample_total}
+
+        before do
+          allow(STDOUT).to receive(:puts).and_return("")
+          allow(dummy_basket).to receive(:itemised_bill).and_return sample_bill
+          allow(sample_bill).to receive(:each)
+          allow(dummy_basket).to receive(:total_bill).and_return sample_total
+        end
+
+        it 'will retrieve the itemised_bill' do
+          expect(dummy_basket).to receive(:itemised_bill).and_return sample_bill
+          menu.review_order
+        end
+
+        it 'will retrieve the total_bill' do
+          expect(dummy_basket).to receive(:total_bill).and_return sample_total
+          menu.review_order
+        end
+      end
+
+
+      describe "#confirm_order" do
+        let(:price) {double :price}
+        let(:sample_total) {double :sample_total}
+        let(:sample_selection) {[[:Carbonara, 3], [:Calzone, 1], [:Risotto, 3]]}
+
+
+        before do
+          allow(dummy_basket).to receive(:total_bill).and_return price
+          allow(dummy_basket).to receive(:checkout)
+          allow(dummy_basket).to receive(:itemised_bill).and_return sample_selection
+          allow(checkout_message).to receive(:send_sms).with(message_text).and_return checkout_message
```

Comment:
Metrics/LineLength: Line is too long. [102/80] (https://github.com/bbatsov/ruby-style-guide#80-character-limits)